src/data/structure/Tuple2.java

```diff
@@ -0,0 +1,26 @@
+package data.structure;
+
+/**
+ * Supports simple pairs of arbitrary items
+ * @author benjaminposerow
+ *
+ * @param <T>
+ * @param <U>
+ */
+public class Tuple2<T,U> {
```

Comment:
What about creating either a sub-package or a separate package for these generic-type structures? data.structure as it currently stands is for the domain model. Maybe we could even rename it.